src/catalog/postgres/builder.cpp

```diff
@@ -548,4 +589,136 @@ IndexSchema Builder::GetConstraintForeignTableIndexSchema(db_oid_t db) {
+  columns.back().SetOid(PRONAMESPACE_COL_OID);
+
+  columns.emplace_back("prolang", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PROLANG_COL_OID);
+
+  columns.emplace_back("procost", type::TypeId::DECIMAL, true, MakeNull(type::TypeId::DECIMAL));
+  columns.back().SetOid(PROCOST_COL_OID);
+
+  columns.emplace_back("prorows", type::TypeId::DECIMAL, true, MakeNull(type::TypeId::DECIMAL));
+  columns.back().SetOid(PROROWS_COL_OID);
+
+  columns.emplace_back("provariadic", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PROVARIADIC_COL_OID);
+
+  columns.emplace_back("proisagg", type::TypeId::BOOLEAN, false, MakeNull(type::TypeId::BOOLEAN));
+  columns.back().SetOid(PROISAGG_COL_OID);
+
+  columns.emplace_back("proiswindow", type::TypeId::BOOLEAN, false, MakeNull(type::TypeId::BOOLEAN));
+  columns.back().SetOid(PROISWINDOW_COL_OID);
+
+  columns.emplace_back("proisstrict", type::TypeId::BOOLEAN, false, MakeNull(type::TypeId::BOOLEAN));
+  columns.back().SetOid(PROISSTRICT_COL_OID);
+
+  columns.emplace_back("proretset", type::TypeId::BOOLEAN, false, MakeNull(type::TypeId::BOOLEAN));
+  columns.back().SetOid(PRORETSET_COL_OID);
+
+  columns.emplace_back("provolatile", type::TypeId::BOOLEAN, false, MakeNull(type::TypeId::BOOLEAN));
+  columns.back().SetOid(PROVOLATILE_COL_OID);
+
+  columns.emplace_back("pronargs", type::TypeId::SMALLINT, false, MakeNull(type::TypeId::TINYINT));
+  columns.back().SetOid(PRONARGS_COL_OID);
+
+  columns.emplace_back("pronargdefaults", type::TypeId::SMALLINT, false, MakeNull(type::TypeId::TINYINT));
+  columns.back().SetOid(PRONARGDEFAULTS_COL_OID);
+
+  columns.emplace_back("prorettype", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PRORETTYPE_COL_OID);
+
+  columns.emplace_back("proargtypes", type::TypeId::INTEGER, false, MakeNull(type::TypeId::INTEGER));
+  columns.back().SetOid(PROARGTYPES_COL_OID);
+
+  columns.emplace_back("proallargtypes", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROALLARGTYPES_COL_OID);
+
+  columns.emplace_back("proargmodes", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROARGMODES_COL_OID);
+
+  columns.emplace_back("proargdefaults", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARCHAR));
+  columns.back().SetOid(PROARGDEFAULTS_COL_OID);
+
+  columns.emplace_back("proargnames", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
+  columns.back().SetOid(PROARGNAMES_COL_OID);
+
+  columns.emplace_back("prosrc", type::TypeId::VARCHAR, 4096, false, MakeNull(type::TypeId::VARCHAR));
+  columns.back().SetOid(PROSRC_COL_OID);
+
+  columns.emplace_back("proconfig", type::TypeId::VARBINARY, 4096, false, MakeNull(type::TypeId::VARBINARY));
```

Comment:
Magic number (4096).  This should probably be a constant defined in `pg_proc.h`.

Reply:
This seems to be an issue for the other builders as well in this file.